src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotationMap.java

```diff
@@ -0,0 +1,208 @@
+package org.broadinstitute.hellbender.tools.funcotator;
+
+import htsjdk.variant.variantcontext.Allele;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.TableFuncotation;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation;
+import org.broadinstitute.hellbender.tools.funcotator.vcfOutput.VcfOutputRenderer;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+
+/**
+ * A linked map of transcript IDs to funcotations.  Also, supports some querying.
+ *
+ * Supports ordering of the transcripts.
+ *
+ * Multiple GencodeFuncotations for the same transcript are prohibited.
+ *
+ * Not thread-safe.
+ */
+public class FuncotationMap {
+
+    public final static String NO_TRANSCRIPT_AVAILABLE_KEY = "no_transcript";
+
+    /** Standard Logger.  */
+    protected static final Logger logger = LogManager.getLogger(FuncotationMap.class);
+
+
+    final private Map<String, LinkedHashSet<Funcotation>> txToFuncotations = new LinkedHashMap<>();
+
+    private FuncotationMap() {}
+
+    private static boolean isGencodeFuncotation(final Funcotation f) {
```

Comment:
This seems like it would make more sense in `FuncotatorUtils`

Reply:
Done.  Added tests, too.